src/include/common/vector/auxiliary_buffer.h

```diff
@@ -43,6 +44,14 @@ class StructAuxiliaryBuffer : public AuxiliaryBuffer {
     std::vector<std::shared_ptr<ValueVector>> childrenVectors;
 };
 
+class ArrowColumnAuxiliaryBuffer : public AuxiliaryBuffer {
```

Comment:
I think the best way to implement the arrow auxiliary is that we store the arrow array in the aux to keep its lifetime, and point to the aux array when possible from ValueVector. Let's discuss this a bit more offline.